example/src/App.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  Button,
+  ScrollView,
+  Platform,
+} from 'react-native';
+import TeadsAdView from '../../src/teads-ad-view';
+import Teads from '../../src/teads';
+import TeadsAdPlacementSettings from '../../src/teads-ad-placement-settings';
+import TeadsAdRequestSettings from '../../src/teads-ad-request-settings';
+import TeadsAdRatio from '../../src/teads-ad-ratio';
+import { Dimensions } from 'react-native';
+import TeadsInReadAdPlacement from '../../src/teads-inread-ad-placement';
+import { DeviceEventEmitter } from 'react-native';
+import { NativeEventEmitter, NativeModules } from 'react-native';
+
+export default function App() {
+  //const testTeadsAdViewEvents = new TeadsAdViewEvents();
+
+  const [showAd, setShowAd] = React.useState<boolean>(true);
+  const [adId, setAdId] = React.useState<String>();
+  const [height, setHeight] = React.useState<number>();
+  const [placement, setPlacement] = React.useState<
+    TeadsInReadAdPlacement | undefined
+  >();
+
+  const testAdPlacementSetting = new TeadsAdPlacementSettings();
+  const testAdRequestSettings = new TeadsAdRequestSettings();
+  const testAdRatio = new TeadsAdRatio(Dimensions.get('window').width);
+
+  React.useEffect(() => {
+    (async () => {
+      await testAdPlacementSetting.enableDebug();
+      await testAdRequestSettings.enableValidationMode();
+      const awaitedVal = await Teads.createInReadPlacement(
+        84242,
+        testAdPlacementSetting
+      );
+      setPlacement(awaitedVal);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  async function onPress() {
+    const resultId = await placement?.requestAd(testAdRequestSettings);
+    setAdId(resultId);
+
+    if (resultId) {
+      testAdRatio.calculateHeight(testAdRatio.width, resultId).then(setHeight);
+    }
+  }
+
+  async function toggleDisplay() {
+    setShowAd(!showAd);
+  }
+
+  async function getEvents() {
+    const eventEmitter = new NativeEventEmitter(NativeModules.RNHandlerEvents);
+    // Subscribe to the event
+    eventEmitter.addListener('didRecordImpression', (event) => {
+      const { adId } = event;
+      console.log('Data received in React on didRecordImpression:', adId);
+    });
+  }
```

Comment:
this part seems very promising, the part that makes me uncomfortable is that you are using some sort of singleton meaning that if you are displaying multiple Teads ads on the same page you need to compare with the `adId` provided I'd prefer to subscribe to events sent on the ad object directly. We can talk about it together if you prefer